From ccb7feb045f7326ee109ba3847b822e5f9ad9381 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:31:36 -0400
Subject: [scsi] hpsa: increase the probability of a reported success after a device reset

Message-id: <1410186734-19986-46-git-send-email-thenzl@redhat.com>
Patchwork-id: 92062
O-Subject: [RHEL6.6 e-stor PATCH 45/83] hpsa: increase the probability of a reported success after a device reset
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

rc is set in the loop, and it isn't set back to zero anywhere
this patch fixes it

Signed-off-by: Tomas Henzl <thenzl@redhat.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
(cherry picked from commit 8919358e3d3e78b97490d9d6b0179685962df58f)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index 769846b..0139364 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -4160,7 +4160,7 @@ static int hpsa_register_scsi(struct ctlr_info *h)
 static int wait_for_device_to_become_ready(struct ctlr_info *h,
  unsigned char lunaddr[])
 {
- int rc = 0;
+ int rc;
  int count = 0;
  int waittime = 1; /* seconds */
  struct CommandList *c;
@@ -4180,6 +4180,7 @@ static int wait_for_device_to_become_ready(struct ctlr_info *h,
    */
   msleep(1000 * waittime);
   count++;
+  rc = 0; /* Device ready. */
 
   /* Increase wait time with each try, up to a point. */
   if (waittime < HPSA_MAX_WAIT_INTERVAL_SECS)
-- 
1.7.1